File: packages/templates/app/python/sample/ethereum.py
from typing import Any, Optional
from polywrap import Client, Uri, PolywrapClient, PolywrapClientConfigBuilder, web3_bundle, sys_bundle

from .wrap import BaseEthereum


class Ethereum(BaseEthereum):
    def _get_default_client(self) -> Client:
        builder = PolywrapClientConfigBuilder()
        builder.add_bundle(web3_bundle)
        config = builder.build()
        return PolywrapClient(config)

    def _get_default_uri(self) -> Optional[Uri]:
        return None

    def _get_default_env(self) -> Any:
        return None

File: packages/templates/app/typescript/src/ethereum.ts
import { BaseEthereum } from "./wrap";

import { CoreClient, PolywrapClient } from "@polywrap/client-js";

export class Ethereum extends BaseEthereum {
  protected _getDefaultClient(): CoreClient {
    return new PolywrapClient();
  }
  protected _getDefaultUri(): string | undefined {
    return undefined;
  }
  protected _getDefaultEnv(): Record<string, unknown> | undefined {
    return undefined;
  }
}
